src/compiler/scala/tools/nsc/transform/CleanUp.scala

```diff
@@ -398,12 +398,15 @@ abstract class CleanUp extends Statics with Transform with ast.TreeDSL {
 
     // transform scrutinee of all matches to ints
     def transformSwitch(sw: Match): Tree = { import CODE._
-      sw.selector.tpe match {
+      sw.selector.tpe.widen match {
         case IntTpe => sw // can switch directly on ints
         case StringTpe =>
           // these assumptions about the shape of the tree are justified by the codegen in MatchOptimization
-          val Match(Typed(selTree: Ident, _), cases) = sw
-          val sel = selTree.symbol
+          val Match(Typed(selTree, _), cases) = sw
+          val selArg = selTree match {
```

Comment:
I'm not sure if this should be a `def` to avoid structural sharing in the AST. Since trees have mutable fields (`.tpe`, `.pos`) this might be a problem - I'm really not sure. @retronym what's your intuition?

Reply:
Oh!  Wow, yeah, good catch, thanks.  Let's do that.